CSC8503CoreClasses/GuardObject.cpp

```diff
@@ -113,8 +112,8 @@ BehaviourAction* GuardObject::ChasePlayerSetup() {
 				this->GetPhysicsObject()->AddForce(Vector3(direction.x, 0, direction.z));
 				mPlayer = dynamic_cast<PlayerObject*>(mPlayer);
 				float dist = (direction.x * direction.x) + (direction.y * direction.y) + (direction.z * direction.z);
-				if (dist < 5) {
-					mHasCaughtPlayer == true;
+				if (sqrtf(dist) < 5) {
```

Comment:
make sure to put magic numbers in variables
consider using the squared distance.